File: WIP - Progetto/app/src/main/java/it/wip/database/model/Shopped.kt
package it.wip.database.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey

@Entity(tableName = "shopped",
        primaryKeys = ["user", "shop_element"],
        foreignKeys = [
            ForeignKey(
                entity = User::class,
                parentColumns = ["id"],
                childColumns = ["user"],
                onUpdate = ForeignKey.CASCADE,
                onDelete = ForeignKey.CASCADE),
            ForeignKey(
                entity = ShopElement::class,
                parentColumns = ["element_name"],
                childColumns = ["shop_element"],
                onUpdate = ForeignKey.CASCADE,
                onDelete = ForeignKey.CASCADE
            )
        ])
data class Shopped(
    val user: Int,
    @ColumnInfo(name = "shop_element")
    val shopElement: String,
    @ColumnInfo(name = "bought_on")
    val boughtOn: String
)

File: wip_flutter/lib/database/model/shopped.dart
class Shopped {

  final int user;
  final String shopElement;
  final String boughtOn;

  Shopped({
    required this.user,
    required this.shopElement,
    required this.boughtOn
  });

  Map<String, dynamic> toMap() {
    return {
      'user': user,
      'shop_element': shopElement,
      'bought_on': boughtOn
    };
  }

}